ethers-core/src/utils/mod.rs

```diff
@@ -954,33 +927,38 @@ mod tests {
 
     #[test]
     fn test_eip1559_default_estimator() {
-        // If the base fee is below the triggering base fee, we should get the default priority fee
-        // with the base fee surged.
-        let base_fee_per_gas = U256::from(EIP1559_FEE_ESTIMATION_PRIORITY_FEE_TRIGGER) - 1;
-        let rewards: Vec<Vec<U256>> = vec![vec![]];
-        let (base_fee, priority_fee) = eip1559_default_estimator(base_fee_per_gas, rewards);
+        // if estimate_priority_fee returns 0, we should return the default priority fee
+        let rewards: Vec<Vec<U256>> = vec![vec![U256::zero()]];
+        let (base_fee, priority_fee) = eip1559_default_estimator(U256::zero(), rewards);
+        assert_eq!(
+            base_fee,
+            U256::from(EIP1559_FEE_ESTIMATION_DEFAULT_PRIORITY_FEE) +
+                U256::from(EIP1559_FEE_ESTIMATION_DEFAULT_BASE_FEE)
+        );
         assert_eq!(priority_fee, U256::from(EIP1559_FEE_ESTIMATION_DEFAULT_PRIORITY_FEE));
-        assert_eq!(base_fee, base_fee_surged(base_fee_per_gas));
 
-        // If the base fee is above the triggering base fee, we calculate the priority fee using
-        // the fee history (rewards).
-        let base_fee_per_gas = U256::from(EIP1559_FEE_ESTIMATION_PRIORITY_FEE_TRIGGER) + 1;
+        // calculate the priority fee using the fee history (rewards).
+        let base_fee_per_gas = U256::from(100_000_000_000u64) + 1;
         let rewards: Vec<Vec<U256>> = vec![
             vec![100_000_000_000u64.into()],
             vec![105_000_000_000u64.into()],
             vec![102_000_000_000u64.into()],
         ]; // say, last 3 blocks
-        let (base_fee, priority_fee) = eip1559_default_estimator(base_fee_per_gas, rewards.clone());
-        assert_eq!(base_fee, base_fee_surged(base_fee_per_gas));
+        let (max_fee, priority_fee) = eip1559_default_estimator(base_fee_per_gas, rewards.clone());
         assert_eq!(priority_fee, estimate_priority_fee(rewards.clone()));
+        let expected_max_fee = base_fee_surged(base_fee_per_gas) + priority_fee;
+        assert_eq!(max_fee, expected_max_fee);
```

Comment:
would also add an assertion on the `max_fee` using a u64 literal to make sure nothing crazy is being returned by `base_fee_surged`

Reply:
Added